diff --git a/src/httpd.c b/src/httpd.c
index 29a347d5..94fb19ba 100644
--- a/src/httpd.c
+++ b/src/httpd.c
@@ -1431,6 +1431,9 @@ httpd_send_reply(struct evhttp_request *req, int code, const char *reason, struc
   if (allow_origin)
     evhttp_add_header(output_headers, "Access-Control-Allow-Origin", allow_origin);
 
+  if(evbuf)
+    evbuffer_add(evbuf, "\r\n", 2);
+
   if (do_gzip && (gzbuf = httpd_gzip_deflate(evbuf)))
     {
       DPRINTF(E_DBG, L_HTTPD, "Gzipping response\n");
diff --git a/src/httpd_daap.c b/src/httpd_daap.c
index 37721c40..65698e1b 100644
--- a/src/httpd_daap.c
+++ b/src/httpd_daap.c
@@ -2439,7 +2439,7 @@ daap_init(void)
   int i;
   int ret;
 
-  srand((unsigned)time(NULL));
+  srand(0);
   current_rev = 2;
   update_requests = NULL;
 
diff --git a/src/main.c b/src/main.c
index 58f12f56..d83e45b4 100644
--- a/src/main.c
+++ b/src/main.c
@@ -235,7 +235,7 @@ register_services(char *ffid, bool no_web, bool no_rsp, bool no_daap, bool no_mp
   int i;
   int ret;
 
-  srand((unsigned int)time(NULL));
+  srand(0);
 
   lib = cfg_getsec(cfg, "library");
 
diff --git a/src/misc.c b/src/misc.c
index 5a849136..0bb26dcb 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1384,7 +1384,7 @@ mutex_init(pthread_mutex_t *mutex)
   int err;
 
   CHECK_ERR(L_MISC, pthread_mutexattr_init(&mattr));
-  CHECK_ERR(L_MISC, pthread_mutexattr_settype(&mattr, PTHREAD_MUTEX_ERRORCHECK));
+  //CHECK_ERR(L_MISC, pthread_mutexattr_settype(&mattr, PTHREAD_MUTEX_ERRORCHECK));
   err = pthread_mutex_init(mutex, &mattr);
   CHECK_ERR(L_MISC, pthread_mutexattr_destroy(&mattr));
 
diff --git a/src/transcode.c b/src/transcode.c
index 77bc9b46..d1985c00 100644
--- a/src/transcode.c
+++ b/src/transcode.c
@@ -772,6 +772,7 @@ open_decoder(unsigned int *stream_index, struct decode_ctx *ctx, enum AVMediaTyp
       avcodec_free_context(&dec_ctx);
       return NULL;
     }
+  dec_ctx->thread_count=1;
 
   if (type == AVMEDIA_TYPE_AUDIO)
     {
